launcher/src/main/java/io/snappydata/tools/QuickLauncher.java

```diff
@@ -387,6 +387,10 @@ private String getHostNameAndDir() {
     commandLine.add("-Dorg.apache.commons.logging.Log=" +
         "org.apache.commons.logging.impl.Log4JLogger");
     setDefaultVMArgs(options, hostData, commandLine);
+    // For lead node set the member-weight to be 17
+    if (launcherClass.equals("io.snappydata.tools.LeaderLauncher")) {
+      vmArgs.add("-Dgemfire.member-weight=17");
+    }
```

Comment:
Since you have added to LeaderLauncher, why is this required?

Also if you keep this too for some reason then can you add a short unit test somewhere to check this weight (maybe just a check in an existing dunit test)? In future this namespace can change from io.snappydata and its likely that changing this may be missed.

Reply:
Tested it manually was working. All the scripts use QuickLauncher which extends LauncherBase calling setDefaultVMArgs.

Will add a test.

Reply:
Ok but you have already this in the override of LeaderLauncher.setDefaultVMArgs, so why is this also required here?

Reply:
LeaderLauncher extends LauncherBase and QuickLauncher extends LauncherBase too.
So any one starting with LeaderLauncher without QuickLauncher would need this like SnappyUtilLauncher.

Reply:
Tested it in a dunit test, the property is not reflected? Checking why.

Reply:
Ultimately QuickLauncher invokes the "launcherClass" i.e. LeaderLauncher so setting as vmArgs inside it will be too late but can be set as system property. I think the cleaner way is to set as system property in LeadImpl.start

Reply:
Quicklauncher invokes ""launcherClass" but uses the vmargs of its own. 
For dunit, will have to make change in LeaderImpl.

Reply:
Why can't you set as a system property in LeadImpl?

Reply:
Yes doing that. It will not require any other changes.